connection_test.go

```diff
@@ -568,6 +581,13 @@ func TestCancelWithoutSendCancelOnContextCanceled(t *testing.T) {
 				_, _, _, err := raw.Call(ctx, ts.Server(), ts.HostPort(), ts.ServiceName(), "ctxWait", nil, nil)
 				assert.Equal(t, ErrRequestCancelled, err, "client call result")
 
+				if tt.wantCancelRequested {
+					serverStats.Expected.IncCounter("inbound.cancels.requested", ts.Server().StatsTags(), 1)
```

Comment:
Yeah, it's missing the `ValidateExpected()`. However, tried adding it below and the test wouldn't pass, because:
1. the `with_relay` subtest inside `WithTestServer` doesn't propagate the cancellation, thus the metric is never incremented;
2. the `no_relay` subtest insde `WithTestServer` increases the metric, but it happens not immediately upon cancel, only after the function passed into `WithTestServer` is finished executing.

I'm not yet sure how to solve this cleanly. I will dig deeper into this once I have more time.

Reply:
This seems to be working (https://github.com/uber/tchannel-go/pull/912/commits/93cd04f2a76436d186f37b0bc420a5504aec9812). Feel free to suggest a different approach.